src/main/java/com/minecolonies/coremod/colony/managers/EventStructureManager.java

```diff
@@ -87,28 +94,55 @@ public boolean spawnTemporaryStructure(
 
         final World world = colony.getWorld();
 
-        final String backupPath = Structures.SCHEMATICS_PREFIX + STRUCTURE_BACKUP_FOLDER + colony.getID() + colony.getDimension() + targetSpawnPoint.down(3);
-        final BlockPos zeroPos = targetSpawnPoint.subtract(structure.getPrimaryBlockOffset());
+        int y = 3;
+
+        final BlockInfo info = structure.getBlockInfoAsMap().getOrDefault(structure.getPrimaryBlockOffset(), null);
+        if (info.getTileEntityData() != null)
+        {
+            final CompoundNBT teData = structure.getTileEntityData(targetSpawnPoint, structure.getPrimaryBlockOffset());
+            if (teData != null && teData.contains(TAG_BLUEPRINTDATA))
+            {
+                final TileEntity entity = TileEntity.create(info.getTileEntityData());
+                if (entity instanceof IBlueprintDataProvider)
+                {
+                    for (final Map.Entry<BlockPos, List<String>> entry : ((IBlueprintDataProvider) entity).getPositionedTags().entrySet())
+                    {
+                        if (entry.getValue().contains("groundlevel"))
+                        {
+                            y = entry.getKey().getY();
+                        }
+                    }
+                }
+            }
+        }
+
+        final BlockPos spawnPos = targetSpawnPoint.add(0, -y ,0);
+
+        final BlockPos zeroPos = targetSpawnPoint.subtract(structure.getPrimaryBlockOffset()).add(0, -y, 0);
+        final BlockPos cornerPos = new BlockPos(zeroPos.getX() + structure.getSizeX() - 1, zeroPos.getY() + structure.getSizeY(), zeroPos.getZ() + structure.getSizeZ() - 1);
+
+        final BlockPos anchor = new BlockPos(zeroPos.getX() + structure.getSizeX()/2, zeroPos.getY(), zeroPos.getZ() + structure.getSizeZ()/2);
+
+        final String backupPath = Structures.SCHEMATICS_PREFIX + STRUCTURE_BACKUP_FOLDER + colony.getID() + colony.getDimension() + anchor;
 
         if (!ItemScanTool.saveStructureOnServer(world,
           zeroPos,
-          new BlockPos(zeroPos.getX() + structure.getSizeX() - 1, zeroPos.getY() + structure.getSizeY(), zeroPos.getZ() + structure.getSizeZ() - 1),
+          cornerPos,
           backupPath,
           false))
         {
             // No structure spawn if we didnt successfully save the surroundings before
             Log.getLogger().info("Failed to save schematics for event");
             return false;
         }
+        backupSchematics.put(anchor, eventID);
```

Comment:
how come we're saving now the schematic's middle position as spawnpoint instead of the spawnpoint?

Reply:
because that is what is used by the scan tool code.

Reply:
ok